src/StructTypes.jl

```diff
@@ -686,4 +703,57 @@ mappings will be applied, and the function will be passed the Julia field name.
     return f_applied
 end
 
+"""
+    makeobj(::Type{Any}, obj::Any)
```

Comment:
I wonder if we just call this `construct`; we already have that defined in StructTypes, but that's really what we're doing here, and custom types are still able to overload it if they want.

Reply:
Hmmm, personally I'd find it confusing if this functionality was also named `construct`.